src/drivers/virtio/virtqueue/packed.rs

```diff
@@ -82,13 +83,9 @@ impl WrapCount {
 }
 
 /// Structure which allows to control raw ring and operate easily on it
-///
-/// WARN: NEVER PUSH TO THE RING AFTER DESCRIPTORRING HAS BEEN INITIALIZED AS THIS WILL PROBABLY RESULT IN A
-/// RELOCATION OF THE VECTOR AND HENCE THE DEVICE WILL NO LONGER NO THE RINGS ADDRESS!
 struct DescriptorRing {
 	ring: &'static mut [Descriptor],
-	//ring: Pinned<Vec<Descriptor>>,
-	tkn_ref_ring: Box<[*mut TransferToken]>,
+	tkn_ref_ring: Box<[MaybeUninit<Box<TransferToken>>]>,
```

Comment:
Would it make sense to replace `MaybeUninit<Box<T>>` with `Option<Box<T>>`?
It is the same size due to niches.
It would incur some overhead due to checking when accessing, but properly ensuring the soundness of accessing `MaybeUninit` might be too difficult, no?